frontend/src/store/plugins/socketPlugin.js

```diff
@@ -0,0 +1,240 @@
+
+  // handle manager events
+  manager.open = function (fn) {
+    logger.debug('manager opening')
+    store.commit('socket/SET_READY_STATE', 'opening')
+    managerOpen(fn)
+  }
+
+  manager.on('open', () => {
+    logger.debug('manager open')
+    store.commit('socket/SET_READY_STATE', 'open')
+  })
+
+  manager.on('close', reason => {
+    logger.debug('manager closed: %s', reason)
+    store.commit('socket/SET_READY_STATE', 'closed')
+  })
+
+  manager.on('error', err => {
+    logger.error('manager error: %s - %s', err.name, err)
+  })
+
+  manager.on('reconnect', attempt => {
+    logger.debug('manager reconnect attempt %d succeeded', attempt)
+  })
+
+  manager.on('reconnect_attempt', attempt => {
+    logger.debug('io reconnect attempt number %d', attempt)
+  })
+
+  manager.on('reconnect_error', err => {
+    logger.error('io reconnect error: %s - %s', err.name, err.message)
+  })
+
+  manager.on('reconnect_failed', () => {
+    logger.error('io reconnect finally failed')
+  })
+
+  // handle socket event
+  const connect = async () => {
+    try {
+      await userManager.ensureValidToken()
+    } finally {
+      socket.connect()
+    }
+  }
+  const reconnect = () => {
+    const attempts = store.state.socket.backoff.attempts
+    if (attempts < 10) {
+      store.commit('socket/BACKOFF_INCREASE_ATTEMPTS')
+      const delay = store.getters['socket/backoffDuration']
+      setTimeout(connect, delay)
+    } else {
+      store.commit('socket/BACKOFF_RESET')
+    }
+  }
+
+  socket.on('connect', () => {
+    logger.info('socket connected')
+    store.dispatch('socket/onConnect', [socket])
+    store.dispatch('shoots/synchronize')
+  })
+
+  socket.on('disconnect', reason => {
+    switch (reason) {
+      case 'io server disconnect': {
+        logger.debug('socket was forcefully disconnected by the server')
+        reconnect()
+        break
+      }
+      case 'io client disconnect': {
+        logger.debug('socket was manually disconnected by the client')
+        break
+      }
+      default: {
+        logger.debug('socket disconnected: %s', reason)
+        break
+      }
+    }
+    store.dispatch('socket/onDisconnect', [socket, reason])
+  })
+
+  const handleManagerError = err => {
+    logger.error('manager error: %s', err.message)
+  }
+  const handleMiddlewareError = err => {
+    let frameRequestCallback
+    const {
+      message,
+      data: {
+        statusCode = 500,
+        code = 'ERR_SOCKET_MIDDLEWARE',
+        ...data
+      } = {}
+    } = err
+    logger.error('socket connect error: %s (%d %s)', message, statusCode, code)
+    switch (code) {
+      case 'ERR_JWT_TOKEN_REFRESH_REQUIRED': {
+        if (typeof data.exp === 'number') {
+          const expiresIn = data.exp - Math.floor(Date.now() / 1000)
+          const tolerance = 5
+          if (expiresIn > tolerance) {
+            frameRequestCallback = () => userManager.signout(createClockSkewError())
+          }
+        }
+        break
+      }
```

Comment:
No. If I remove the manager.open monkey patch it will not work. We only wanted to signout if the backend says expired but the frontend thinks it should be valid.

Reply:
right, a refresh is still possible. Somehow I misinterpreted the error code at that point in time. -> no issue